### PR TITLE
Parse Schema

### PR DESCRIPTION
**Pull request by pinguineras**
Creating a Class ParseSchema to handle Schemas from Parse and ParseSchemaTest to PHPUnit.
> Get all schemas from Application
> Get specific schema
> Creating a new Schema
> Update a Schema
> Deleting a schema

It's a pleasure to help!

### CODE REVIEW

**Comment by pinguineras** [comment]
Finally checks passed! lol

**Comment by GrahamCampbell** [comment]
You know you could have just downloaded an applied the diff StyleCI gave you. :trollface:

https://styleci.io/analyses/zYdNGq/diff

File: src/Parse/ParseSchema.php [hunk 1]
```diff
@@ -0,0 +1,498 @@
...
+     * @see https://parse.com/docs/rest/guide#schemas
+     * 
+     * @var bool
+     */
+    public static $useMasterKey = true;
+
+    /**
+     * Create a Parse Schema.
+     *
+     * @param mixed $className Class Name of data on Parse.
```
Comment (GrahamCampbell) [inline src/Parse/ParseSchema.php]: it's not really mixed is it?

Comment (GrahamCampbell) [inline src/Parse/ParseSchema.php]: `string|null` maybe?

Comment (pinguineras) [inline src/Parse/ParseSchema.php]: Sure! I'll make the modification. 

Thanks

**Comment by pinguineras** [comment]
lol! You're right again!!! Thanks again!

**Comment by GrahamCampbell** [comment]
Haha. I run StyleCI, so know where I put everything, lol. :)

File: src/Parse/ParseSchema.php [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+            throw new Exception('You must set a Class Name before make any request.');
+        }
+    }
+
+    /**
+     *  Assert types of fields.
+     * 
+     * @throws Exception
+     */
+    public function assetTypes($type = null)
```
Comment (gfosco) [inline src/Parse/ParseSchema.php]: asset => assert

File: tests/Parse/ParseSchemaTest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        $schema->addBoolean('NewField4');
+        $schema->addDate('NewField5');
+        $schema->addFile('NewField6');
+        $schema->addGeoPoint('NewField7');
+        $schema->addArray('NewField8');
+        $schema->addObject('NewField9');
+        $schema->addPointer('NewField10', '_User');
+        $schema->addRelation('NewField11', '_User');
+        $schema->save();
+    }
```
Comment (gfosco) [inline tests/Parse/ParseSchemaTest.php]: Can you get the schema and check that these fields were set properly?

File: tests/Parse/ParseSchemaTest.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    public function testUpdateSchema()
+    {
+        $schema = new ParseSchema('SchemaTest');
+        $schema->deleteField('NewField2');
+        $schema->addNumber('quantity');
+        $schema->addField('status', 'Boolean');
+        $schema->update();
+    }
```
Comment (gfosco) [inline tests/Parse/ParseSchemaTest.php]: Likewise, can you fetch the updated schema and verify the updates took?

File: tests/Parse/ParseSchemaTest.php [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    public function testDeleteSchema()
+    {
+        $schema = new ParseSchema('SchemaTest');
+        $schema->delete();
+    }
```
Comment (gfosco) [inline tests/Parse/ParseSchemaTest.php]: This should create one, then delete, then verify it is gone.

**Comment by gfosco** [comment]
Cool... I added some comments.  Once they're addressed we can merge. :+1:

**Comment by pinguineras** [comment]
Great, i'll made the changes and commit today or tomorrow! Thanks!

**Comment by gfosco** [comment]
:+1:

**Pull request merged**